docker-compose.yml

```diff
@@ -0,0 +1,107 @@
+---
+version: '3.7'
+services:
+
+  imsearch:
+    image: imsearch-cpu:latest
+    container_name: ${NAMESPACE}-cpu
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+    - 5000:5000
+    environment:
+    - MONGO_URI=mongodb://mongodb:27017/
+    - REDIS_URI=redis://redis:6379/0
+    - STORAGE_MODE=local
+    depends_on:
+    - redis
+    - mongodb  
+    links:
+    - mongodb:mongodb
+    - redis:redis
+    networks:
+    - internal
+    - web
+
+  mongodb:
+    image: webhippie/mongodb:latest
+    container_name: ${NAMESPACE}-mongodb
+    restart: always
+    ports:
+    - 27017:27017
+    - 28017:28017
+    - 27019:27019
+    volumes:
+    - mongo-data:/var/lib/mongodb/db
+    - mongo-backup:/var/lib/backup
+    networks:
+    - internal
+    environment:
```

Comment:
I am not sure why these environment variables are required? Can't we get rid of these unwanted variables?

Reply:
If you want to setup an insecure database, we can remove them.

Reply:
I am just confused. What will happen if these environment variables are not provided? I mean what if these are not set anyhow? Also, will `.env` file will automatically loaded while `docker-compose` runs?

Reply:
empty vars gonna be provided, so an open mongodb database will be created

Reply:
Will the default values be overridden?

Reply:
I can change for the default container, just tried to make something security/deployment wise smart.

Reply:
I understand that. And that is totally perfect. I am just trying to understand it better.

Reply:
We can keep the variables as long as id does not affect the behavior other than security if no variables are provided.

Reply:
I am changing for the default mongo:latest, gonna be more simple for other users/devs.

Reply:
I have changed to the normal container, and removed the .env file